File: vlife-admin-react/src/api/Button.ts

import {PageVo,DbEntity,Result, PageQuery} from '@src/api/base'

import apiClient from '@src/api/base/apiClient'
import { BtnToolBarPosition } from '@src/components/button/types';
import { FormRuleDto } from './FormRule';
// 按钮
export interface Button extends DbEntity{
  sysMenuId: string;  // 所在菜单
  formId: string;  // 模型 id
  sysResourcesId: string;  // 接口
  position:BtnToolBarPosition;//位置
  icon: string;  // 图标
  title: string;  // 按钮名称
  submitConfirm: boolean;  // 确认提醒
  disabledHide: boolean;  // 禁用时隐藏
  conditionJson:string;// 按钮可用条件（组装unableMatch函数）
  activeTabKey:string //按钮能运行的页签
  sort: number;  // 排序号
  formVoJson:string;//模型信息
  code:string;//按钮标识
  //------用户不可见字段
  // formTitle: string;  // 按钮表单名称
  // toActiveTabKey: string;  // 完成后去到的场景页签
  // btnType: string;  // 样式
  // tooltip: string;  // 不可用时提示
  // actionType: string;  // 类型
  // model: string;  // 表单模型
  // multiple: boolean;  // 处理多条记录
  // remark: string;  // 按钮说明
}


export interface ButtonVo extends Button{
  rules:FormRuleDto[]
  
}
/** save*/
export const save=(dto:Button): Promise<Result<Button>>=>{
  return apiClient.post(`/button/save`,dto);
};


export const moveUp=(dto:Button): Promise<Result<Button>>=>{
  return apiClient.post(`/button/moveUp`,dto);
};

/** page*/
export const page=(req:PageQuery): Promise<Result<PageVo<Button>>>=>{
  return apiClient.post(`/button/page`,req);
};
/** detail*/
export const detail=(req:{id:string}): Promise<Result<Button>>=>{
  return apiClient.get(`/button/detail/${req.id}`);
};
/** remove*/
export const remove=(ids:string[]): Promise<Result<number>>=>{
  return apiClient.delete(`/button/remove`,{data:ids});
};


export const list=(req?:PageQuery): Promise<Result<ButtonVo[]>>=>{
  return Promise.resolve({ data: [],code: "200", msg: "" }); 
  // return apiClient.post(`/button/list/buttonVo`,req);
};

export const moveDown=(dto:Button): Promise<Result<Button>>=>{
  return apiClient.post(`/button/moveDown`,dto);
};

File: vlife/vlife-base/src/main/java/cn/wwwlike/form/entity/Button.java
package cn.wwwlike.form.entity;

import cn.wwwlike.vlife.annotation.VField;
import cn.wwwlike.vlife.bean.DbEntity;
import lombok.Data;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

/**
 * 按钮
 */
@Setter
@Entity
@Table
public class Button extends DbEntity {
    //按钮唯一编码
    public String code;
    //所在菜单
    public String sysMenuId;
    //模型 id
    public String formId;
    //接口
    public String sysResourcesId;
    //按钮名称
    public String title;
    //图标
    public String icon;
    //使用场景
    @VField(dictCode = "button_position")
    public String position;
    //按钮启用条件
    public String conditionJson;
    //确认提醒
    public boolean submitConfirm;
    //禁用时隐藏
    public boolean disabledHide;
    //按钮表单名称
    public String formTitle;
    //表单模型
    public String model;
    //类型
    public String actionType;
    //样式
    public String btnType;
    //按钮说明
    public String remark;
    //处理多条记录
    public boolean multiple;
    //不可用时提示
    public String tooltip;
    //排序号
    public Integer sort;
    //完成后去到的场景页签
    public String toActiveTabKey;
    /**
     * 所在页签
     * 为空则所有页签可用（id）
     */
    public String activeTabKey;

    public String getCode() {
        return code;
    }

    //表单配置json
    public String formVoJson;
    @Column(columnDefinition = "text")
    public String getFormVoJson() {
        return formVoJson;
    }

    @Column(columnDefinition = "text")
    public String getConditionJson() {
        return conditionJson;
    }

    public String getSysMenuId() {
        return sysMenuId;
    }

    public String getFormId() {
        return formId;
    }

    public String getSysResourcesId() {
        return sysResourcesId;
    }

    public String getTitle() {
        return title;
    }

    public String getFormTitle() {
        return formTitle;
    }

    public String getModel() {
        return model;
    }

    public String getActionType() {
        return actionType;
    }

    public String getBtnType() {
        return btnType;
    }

    public String getIcon() {
        return icon;
    }

    public String getRemark() {
        return remark;
    }

    public boolean isMultiple() {
        return multiple;
    }

    public boolean isSubmitConfirm() {
        return submitConfirm;
    }

    public boolean isDisabledHide() {
        return disabledHide;
    }

    public String getTooltip() {
        return tooltip;
    }

    public Integer getSort() {
        return sort;
    }

    public String getToActiveTabKey() {
        return toActiveTabKey;
    }

    public String getPosition() {
        return position;
    }

    public String getActiveTabKey() {
        return activeTabKey;
    }

}
